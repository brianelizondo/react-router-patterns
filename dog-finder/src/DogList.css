.DogList{
    text-align: center;
}
.DogList-logo{
    padding-bottom: 30px;
}
.DogList-logo img{
    max-width: 250px;
}
.DogList-list a{
    color: black;
    text-decoration: none;
}
.DogList-list a:hover{
    color: turquoise;
    text-decoration: underline;
}