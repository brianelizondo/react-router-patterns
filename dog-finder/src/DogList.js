import React from "react";
import { Link } from "react-router-dom";
import "./DogList.css"

import dog_logo from "./dog_finder_logo.jpg"

function DogList({ dogs }){
    return (
        <div className="DogList">
            <div className="DogList-logo"><img src={ dog_logo } alt="" /></div>
            <div className="DogList-list">
                <h2>PET's LIST:</h2>
                { dogs.map(dogData => (<div key={ dogData.name.toLowerCase() }><Link to={ `/dogs/${ dogData.name.toLowerCase() }` }>{ dogData.name }</Link></div>)) }
            </div>
        </div>
    );
}

export default DogList;