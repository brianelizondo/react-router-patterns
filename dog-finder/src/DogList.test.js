import React from "react";
import { render } from "@testing-library/react";
import { MemoryRouter } from "react-router-dom";

import DogList from "./DogList";
import whiskey from "./whiskey.jpg";

const dogTest = [{
    name: "Whiskey",
    age: 5,
    src: whiskey,
    facts: [
        "Whiskey loves eating popcorn.",
        "Whiskey is a terrible guard dog.",
        "Whiskey wants to cuddle with you!"
    ]
}];

// Smoke Tests
it("'DogList' component renders without crashing", () => {
    render(<MemoryRouter><DogList dogs={ dogTest } /></MemoryRouter>);
});

// Snapshot Tests
it("'DogList' matches snapshot", () => {
    const {asFragment} = render(<MemoryRouter><DogList dogs={ dogTest } /></MemoryRouter>);
    expect(asFragment()).toMatchSnapshot();
});

// Others test
it("'DogList' works listed the dogs", () => {
    const { queryByText } = render(<MemoryRouter><DogList dogs={ dogTest } /></MemoryRouter>);
  
    // expect list the dog
    expect(queryByText("PET's LIST:")).toBeInTheDocument();
    expect(queryByText("Whiskey")).toBeInTheDocument();
});