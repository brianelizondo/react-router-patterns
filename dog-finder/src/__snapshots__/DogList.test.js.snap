// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`'DogList' matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="DogList"
  >
    <div
      class="DogList-logo"
    >
      <img
        alt=""
        src="dog_finder_logo.jpg"
      />
    </div>
    <div
      class="DogList-list"
    >
      <h2>
        PET's LIST:
      </h2>
      <div>
        <a
          href="/dogs/whiskey"
        >
          Whiskey
        </a>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
